### PR TITLE
Pick up various updates to shorten file names on Windows

### PR DESCRIPTION
**Pull request by dwnusbaum**
See jenkinsci/jenkins-test-harness#113. Filing now to test the upstream PR and so that I actually remember that the tests should be unignored.

### CODE REVIEW

**Comment by dwnusbaum** [comment]
Should have looked more carefully at the upstream PR, the test here doesn't actually use RestartableJenkinsRule so it's unaffected by that change.

**Comment by dwnusbaum** [comment]
I'll try to tweak [AbstractSampleDVCSRepoRule](https://github.com/jenkinsci/scm-api-plugin/blob/089c807b677991a57ec48e522c823ce452e597de/src/test/java/jenkins/scm/impl/mock/AbstractSampleDVCSRepoRule.java#L51) as suggested by Jesse in https://github.com/jenkinsci/workflow-multibranch-plugin/pull/82#issuecomment-427401886 to see if that helps.

File: pom.xml [hunk 1]
```diff
@@ -64,13 +64,14 @@
...
-        <jenkins.version>2.73.3</jenkins.version>
+        <jenkins.version>2.107.3</jenkins.version>
         <java.level>8</java.level>
         <workflow-cps-plugin.version>2.57</workflow-cps-plugin.version>
         <git-plugin.version>3.6.4</git-plugin.version>
-        <scm-api-plugin.version>2.2.5</scm-api-plugin.version>
+        <scm-api-plugin.version>2.2.7</scm-api-plugin.version>
         <workflow-scm-step-plugin.version>2.4</workflow-scm-step-plugin.version>
         <workflow-multibranch-plugin.version>2.10</workflow-multibranch-plugin.version>
+        <jenkins-test-harness.version>2.44-20181005.160621-2</jenkins-test-harness.version> <!-- TODO https://github.com/jenkinsci/jenkins-test-harness/pull/113 -->
```
Comment (jglick) [inline pom.xml]: pending https://github.com/jenkinsci/plugin-pom/pull/126

Comment (jglick) [reply pom.xml]: now use parent 3.25

**Pull request merged**